### DISCUSSION TITLE
Runtime filterable distinctAttributes

### DISCUSSION THREAD

**Discussion by ilionic** [meilisearch/product#587] [Feedback & Feature Proposal]
### Problem

At the moment it's only possible to set `distinctAttributes` as an index setting.

### Solution / Use cases

To make `distinctAttributes` work as a filter / query similar to SQL `distinct_on` or `distinct` override in [Algolia](https://www.algolia.com/doc/api-reference/api-parameters/distinct/?client=javascript) 

There are many use cases, for example in e-commerce to have bundles of limited edition products in one view ( with "hide copies" filter enabled ) OR having full list of items ( with "hide copies" disabled ).

**Comment by gmourier** [comment]
Hello @ilionic 👋

It seems to be related to #414. Am I right?

Please let me know so that I can centralize both discussions.

Thank you!

**Reply by ilionic** [reply]
Yes, looks like that's the case. Thanks!

**Comment by macraig** [comment]
Closing as defining distinct attributes at search time has been implemented in [v1.9](https://github.com/meilisearch/meilisearch/releases/tag/v1.9.0) 🦎 

Happy searching ✨
